### PR TITLE
feat(patients): add ability to create new patient

### LINKED ISSUE #1671: Create New Patient
## Scenario
**AS** a user with patient write access
**I WANT** to be able to create a new patient
**SO THAT** I can enter a new patient into my system

## Acceptance Criteria
**GIVEN** there is a new patient to be added to the system
**WHEN** I access the /new/patients page
**THEN** I should see a form that allows me to enter the following data: 

* Name - (prefix, given name, family name, suffix)
* Sex
* Date of Birth
* Approximate Age
* Patient Type
* Occupation
* Preferred Language
* Phone Number
* Email
* Address

**GIVEN** a new patient form
**WHEN** I select the approximate age checkbox
**THEN** A new text box appears that allows me to enter an approximate age
**AND THEN** when I enter the approximate age, the date of birth field should be populated with the date which equals current date - approximate age.

## Technical Notes
https://www.hl7.org/fhir/patient.html

## Related Issues
* #1612

### PR DESCRIPTION
**Pull request by jackcmeyer**
Fixes #1671 #1612 .

**Changes proposed in this pull request:**
- adds ability to create a new patient
- adds ability to approximate age

**Newly added dependencies with [Bundlephobia](https://bundlephobia.com/) links:**

- [date-fns](https://bundlephobia.com/result?p=date-fns@2.8.1)

### CODE REVIEW

File: src/components/Navbar.tsx [hunk 1]
```diff
@@ -32,7 +31,7 @@ const Navbar = () => {
             {
               label: t('actions.new'),
               onClick: () => {
-                history.push('/patients/new')
+                history.push('/new/patients')
```
Comment (fox1t) [inline src/components/Navbar.tsx]: Why changing this line? I think it is better to got with `/entityName(s)/new` for every entity we have

File: src/model/Patient.ts [hunk 2]
```diff
@@ -1,13 +1,51 @@
...
+  preferredLanguage?: string
+
+  occupation?: string
+
+  type?: string
+
+  constructor(
+    id: string,
+    rev: string,
+    name: Name,
```
Comment (fox1t) [inline src/model/Patient.ts]: I am not sure we need a custom class for name. Another problem in having 2 depth prop is that we have to check if `.name` exists before doing `.name.given`. I think it is better to flat it.

File: src/components/input/DatePickerWithLabelFormGroup.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  value: string
+  isEditable: boolean
+  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void
+}
+
+const DatePickerWithLabelFormGroup = (props: Props) => {
+  const { onChange, label, name, value, isEditable } = props
+  const id = `${name}DatePicker`
+  return (
+    <div className="form-group">
```
Comment (fox1t) [inline src/components/input/DatePickerWithLabelFormGroup.tsx]: Why not using our date picker component here?

File: src/model/Name.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+export default class Name {
```
Comment (fox1t) [inline src/model/Name.ts]: Not sure we need this class.

**Comment by fox1t** [comment]
This PR is not passing tests at the moment: `> 47 |     history.push(`patients/${newPatient.id}`)` produces `Attempted to log "TypeError: Cannot read property 'body' of null`

File: package.json [hunk 5]
```diff
@@ -10,10 +10,12 @@
     "@semantic-release/changelog": "~3.0.4",
     "@semantic-release/git": "~7.0.16",
     "@semantic-release/release-notes-generator": "~7.3.0",
+    "@types/luxon": "^1.21.0",
     "bootstrap": "~4.4.1",
     "i18next": "^19.0.1",
     "i18next-browser-languagedetector": "^4.0.1",
     "i18next-xhr-backend": "^3.2.2",
+    "luxon": "^1.21.3",
```
[the remark this answers is not in the document]
Comment (jackcmeyer) [reply package.json]: For no good reason other than familiarity with luxon/moment. 

I can investigate using `date-fns`

File: src/components/Navbar.tsx (hunk 1 above)
Comment (jackcmeyer) [reply src/components/Navbar.tsx]: I'm considered that if we have route such as 

`/entity/:id`
and 
`/entity/new`

That we could end up with collisions. 

This guarantees that we will never have that happen.

File: src/components/input/DatePickerWithLabelFormGroup.tsx (hunk 3 above)
Comment (jackcmeyer) [reply src/components/input/DatePickerWithLabelFormGroup.tsx]: updated to use our date picker

File: src/components/Navbar.tsx (hunk 1 above)
Comment (jackcmeyer) [reply src/components/Navbar.tsx]: I updated the route back to what it was. 😊

File: src/model/Name.ts (hunk 4 above)
Comment (jackcmeyer) [reply src/model/Name.ts]: Removed this class.

File: src/model/Patient.ts (hunk 2 above)
Comment (jackcmeyer) [reply src/model/Patient.ts]: Changed it to flatten the name object.

File: package.json (hunk 5 above)
[the remark this answers is not in the document]
Comment (jackcmeyer) [reply package.json]: Switched to use date-fns. First time using it, I enjoyed their API. 🎉

**Pull request merged**